### ISSUE TITLE
Treat URLs to known licenses the same as full text license

### ISSUE THREAD

**Issue by jhansche** [licensee/licensee#434]
### Is your feature request related to a problem? Please describe the problem you're trying to solve.

If a LICENSE file or similar entry consists solely of a link to a well known license URL (such as `http://www.apache.org/licenses/LICENSE-2.0.txt`), this is currently not detected as the "Apache-2.0" license.

Some dependencies declare their license intent by just a plain URL to the license (whether it's a proprietary license or a well known open source license). Currently a file with this content is detected as `None` or `NOASSERTION`.

### Describe the solution you'd like

A license described only by its URL, at least for well-known license URLs like this, should be interpreted as the associated license.  Such that `http://www.apache.org/licenses/LICENSE-2.0.txt` should be interpreted as `Apache-2.0` license.

### Describe alternatives you've considered

One workaround could be to detect if the contents of the file consist only of a URL, and then download that URL into a LICENSE file in order to match against it. But matching against a URL is much less text to sift through, compared to the full text of the license.

### Additional context

This particular project is using the [Google Open Source Licenses plugin for Gradle](https://developers.google.com/android/guides/opensource) in order to extract license information from each declared Gradle dependency. That content is then written to a seek-file with separate offset/length metadata. The contents can be extracted from the seek file by reading the start/length range from the file, for each dependency.  Some of those extracted licenses are full text (in some cases, the full Apache-2 license, or the Android SDK license, etc); but many of those are declared using only the license URL -- some of which are links to popular licenses like Apache-2.0, though some are considered proprietary licenses pointing to that company's terms of services.

Examples:
* Dagger (`com.google.dagger:dagger`) dependency extracts the full Apache-2.0 license text
* Kotlin Coroutines (`org.jetbrains.kotlinx:kotlinx-coroutines-core`) extracts only the URL: `http://www.apache.org/licenses/LICENSE-2.0.txt`
* Fabric plugin (`io.fabric.sdk.android:fabric`) extracts a URL to the company's terms: `https://fabric.io/terms`

The [kotlinx-coroutines POM](https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-core/1.3.5/kotlinx-coroutines-core-1.3.5.pom) defines the license as:
```xml
  <licenses>
    <license>
      <name>The Apache Software License, Version 2.0</name>
      <url>https://www.apache.org/licenses/LICENSE-2.0.txt</url>
      <distribution>repo</distribution>
    </license>
  </licenses>
```
But the Google `oss-licenses-plugin` extracts only the URL from this.

**Comment by mlinksva** [comment]
I'm dubious about treating a well known URL in a `LICENSE` file as equivalent to the license text. I've never seen that practice, and it's not clear to me it'd be a clarifying one given the requirements of most OSS licenses to keep a copy of the license.

NuGet package manager file support was just added in #432 with matching for certain well known license URLs in license fields supported by NuGet. If there are other package managers where it is common to identify licenses with well known URLs, similar support could be added to those (or included when support for the package manager is added to licensee; I don't think the examples you give currently are).

Closing this as I don't think there's anything immediately actionable, but feel free to continue to discuss, or open narrower issues or PRs.

**Comment by jhansche** [comment]
From reading the SPDX spec, one of the fields in the index is a permanent canonical URL. So it stands to reason that a matcher should reasonably be able to infer the license from that canonical URL.

ScanCode for example (a tool written in Python that has similar license matching) is able to infer the Apache-2.0 license, using only the URL to match against.

I'm confused why this would need to be closed, because in your own comments to #432, you say
> Fetching the content specified by `licenseUrl` would be out of scope for licensee IMO, but extracting license identification from well known URLs seems within scope for licensee.

Isn't that what I'm asking for?
